pyomo/core/base/component.py

```diff
@@ -128,6 +129,25 @@ def __deepcopy__(self, memo):
                 ans = memo[id(self)] = self
                 return ans
 
+        #
```

Comment:
It doesn't look like these updates made it into the __deepcopy__ method in kernel/component_interface.py (which is a copy-paste of this method)

Reply:
I tried copying over the current implementation of `__deepcopy__`, and I am getting errors from Kernel's `__setstate__` that are not obvious (to me).  For example:
```
======================================================================
ERROR: test suite for <class 'pyomo.core.tests.unit.test_component_block.Test_block_clone'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "nose/1.3.7/lib/python2.7/site-packages/nose/suite.py", line 209, in run
    self.setUp()
  File "nose/1.3.7/lib/python2.7/site-packages/nose/suite.py", line 292, in setUp
    self.setupContext(ancestor)
  File "nose/1.3.7/lib/python2.7/site-packages/nose/suite.py", line 315, in setupContext
    try_run(context, names)
  File "nose/1.3.7/lib/python2.7/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "pyomo/core/tests/unit/test_component_block.py", line 1622, in setUpClass
    model = cls._block = model.clone()
  File "pyomo/core/kernel/component_block.py", line 106, in clone
    '__paranoid__': True,
  File "python/2.7.14/lib/python2.7/copy.py", line 174, in deepcopy
    y = copier(memo)
  File "pyomo/core/kernel/component_interface.py", line 289, in __deepcopy__
    ans.__setstate__(new_state)
  File "pyomo/core/kernel/component_interface.py", line 322, in __setstate__
    if self._parent is not None:
AttributeError: 'block' object has no attribute '_parent'
```